src/syrupy/serializers/amber.py

```diff
@@ -163,19 +184,28 @@ def serialize_iterable(
 
     @classmethod
     def serialize_unknown(
-        cls, data: Any, *, depth: int = 0, visited: Set[Any] = set()
+        cls, data: Any, *, depth: int = 0, visited: Optional[Set[Any]] = None
     ) -> str:
         return cls.with_indent(repr(data), depth)
 
     @classmethod
     def serialize(
-        cls, data: "SerializableData", *, depth: int = 0, visited: Set[Any] = set()
+        cls,
+        data: "SerializableData",
+        *,
+        depth: int = 0,
+        visited: Optional[Set[Any]] = None,
     ) -> str:
+        visited = visited if visited is not None else set()
```

Comment:
Wouldn't this also work?
```suggestion
        visited = visited or set()
```

Reply:
no need to create a new set if the set is empty